src/main/resources/services/graphQL/collection/updateCollection.es

```diff
@@ -0,0 +1,134 @@
+	//log.debug(`documentTypeId:${toStr(documentTypeId)}`);
+
+	const writeConnection = connect({
+		principals: [PRINCIPAL_EXPLORER_WRITE]
+	});
+
+	const oldNode = writeConnection.get(_id);
+	//log.debug(`oldNode:${toStr(oldNode)}`);
+
+	if (_name !== oldNode._name) {
+		const moveParams = {
+			source: _id,
+			target: _name
+		};
+		//log.debug(`moveParams:${toStr({moveParams})}`);
+		const boolMoved = writeConnection.move(moveParams);
+		if (!boolMoved) {
+			throw new Error(`Unable to move/rename _id:${_id} from _path:${oldNode._path} to _name:${_name}!`);
+		}
+		writeConnection.refresh(); // So the data becomes immidiately searchable
+	}
+
+	const propertiesToBeUpdated = {
+		//_indexConfig: {default: 'byType'},
+		//_inheritsPermissions: true,
+		//_name,
+		//_nodeType: NT_COLLECTION,
+		//_parentPath: '/collections',
+		//_permissions: [],
+		//creator = getUser().key
+		//createdTime: new Date()
+		modifiedTime: new Date(),
+		modifier: getUser().key,
+		language
+	};
+	//log.debug(`propertiesToBeUpdated:${toStr(propertiesToBeUpdated)}`);
+
+	if (collector) {
+		const {
+			name: collectorName,
+			configJson
+		} = collector;
+		if (collectorName || configJson) {
+			propertiesToBeUpdated.collector = {};
+		}
+		if (collectorName) {
+			propertiesToBeUpdated.collector.name = collectorName;
+		}
+		if (configJson) {
+			propertiesToBeUpdated.collector.configJson = configJson;
+			propertiesToBeUpdated.collector.config = JSON.parse(configJson);
+		}
+	}
+	//log.debug(`propertiesToBeUpdated:${toStr(propertiesToBeUpdated)}`);
+
+	if (!documentTypeId) {
+		// Create DocumentType With CollectionName pluss some integer if already existing
+		let documentTypeName = _name;
+		let i = 0;
+		while (exists({
+			connection: writeConnection,
+			_parentPath: '/documentTypes',
+			_name: documentTypeName
+		})) {
+			i++;
+			documentTypeName = `${_name}_${i}`; // /^[a-z][a-zA-Z0-9_]*$/
+		}
+		const createdDocumentTypeNode = createDocumentType({_name: documentTypeName});
+		//log.debug(`createdDocumentTypeNode:${toStr(createdDocumentTypeNode)}`);
+		documentTypeId = createdDocumentTypeNode._id;
+	}
+	propertiesToBeUpdated.documentTypeId = reference(documentTypeId);
+	//log.debug(`propertiesToBeUpdated:${toStr(propertiesToBeUpdated)}`);
+
+	const modifiedNode = writeConnection.modify({
+		key: _id,
+		editor: (n) => {
+			Object.keys(propertiesToBeUpdated).forEach((property) => {
+				const value = propertiesToBeUpdated[property];
+				n[property] = value;
+			});
+			return n;
+		}
+	});
+	//log.debug(`modifiedNode:${toStr(modifiedNode)}`);
+
+	if (modifiedNode) {
+		writeConnection.refresh(); // So the data becomes immidiately searchable
+		modifiedNode.cron = cron;
+		modifiedNode.doCollect = doCollect;
+		//log.debug(`modifiedNode:${toStr(modifiedNode)}`);
+		createOrModifyJobsFromCollectionNode({
+			connection: writeConnection,
+			collectionNode: modifiedNode,
+			timeZone: 'GMT+02:00' // CEST (Summer Time)
```

Comment:
Hmm, is it correct to hardcode time zone like that? If it's local time zone then here's solution:

https://stackoverflow.com/a/24500441

Reply:
https://github.com/enonic/app-explorer/issues/332